File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2019w/ID07IO1/BridgeTest/src/FigureCircle.java

import java.awt.Color;
import java.awt.Graphics;

public class FigureCircle extends Figure
{
    private double ax,ay,radius;
    private FigureCircle() {}
    public static Figure make(String str)
    {
        double[] tb=toDoubleArray(str);
        if(tb.length!=3) return null;
        FigureCircle fig=new FigureCircle();
        int i=0;
        fig.ax=tb[i++];
        fig.ay=tb[i++];
        fig.radius=tb[i++];
        return fig;
    }
    @Override public String toString() 
    {
        return "Circle{" + ax + ", " + ay + ", " + radius + '}';
    }
    @Override public void draw(Graphics g)
    {
        g.setColor(Color.RED);
        g.drawOval(to_i(ax-radius), to_i(ay-radius), to_i(2*radius), to_i(2*radius));
    }
}

File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2020/ID06TC1/BridgeTest/BridgeTest/FigureCircle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Drawing;

namespace BridgeTest
{
    class FigureCircle : Figure
    {
        private static readonly int PARAM_COUNT=3;
        private int x, y, r;
        public FigureCircle(int[] tb)
        {
            if (tb.Length != PARAM_COUNT) throw new ArgumentException("Expected " + PARAM_COUNT + " paramters");
            x = tb[0];
            y = tb[1];
            r = tb[2];
        }
        public void Draw(Graphics gr)
        {
            gr.DrawEllipse(Pens.Red, x - r, y - r, r << 1, r << 1); // r << 1 <=> r*2;
        }
        public string getKind() { return "Circle"; }
        public string getParameters()
        {
            return string.Format("{0}, {1}, {2}", x, y, r);
        }
        public override string ToString()
        {
            return getKind() + '(' + getParameters() + ')';
        }
    }
}
